File: AllJobsHomeAssignment.Server/Models/Order.cs
using System.ComponentModel.DataAnnotations;

namespace AllJobsHomeAssignment.Server.Models;

public class Order
{
    public int Id { get; set; }

    [Required]
    [MaxLength(255)]
    public string CustomerName { get; set; } = null!;

    [Required]
    [MaxLength(255)]
    public string CustomerEmail { get; set; } = null!;

    public OrderStatus Status { get; set; } = OrderStatus.NEW;

    [Required]
    public decimal TotalAmount { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public ICollection<OrderItem> Items { get; set; } = new List<OrderItem>();
}

File: alljobshomeassignment.client/models/order.ts
export type OrderStatus = 'NEW' | 'PAID' | 'CANCELLED'

export interface OrderItem {
  id?: number
  orderId?: number
  productId: number
  quantity: number
  unitPrice: number
  lineTotal: number
}

export interface Order {
  id: number
  customerName: string
  customerEmail: string
  status: OrderStatus
  totalAmount: number
  items: OrderItem[]
  createdAt: string
  updatedAt: string
}

export interface CreateOrderRequest {
  customerName: string
  customerEmail: string
  items: { productId: number; quantity: number }[]
}
